src/docfx/docset/BuildScope.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal class BuildScope
+    {
+        public Func<string, bool> Glob { get; private set; }
+
+        public HashSet<Document> Files { get; private set; }
+
+        public HashSet<Document> FilesWithFallback { get; private set; }
+
+        public RedirectionMap Redirections { get; private set; }
+
+        public static (List<Error>, BuildScope) Create(Docset docset)
+        {
+            var glob = CreateGlob(docset);
+
+            var (errors, redirections) = RedirectionMap.Create(docset, glob);
+            var files = GetFiles(docset, glob).Concat(redirections.Files).ToHashSet();
+            var filesWithFallback = files;
+
+            if (docset.FallbackDocset != null)
+            {
+                filesWithFallback = files.Concat(
+                    GetFiles(docset.FallbackDocset, CreateGlob(docset.FallbackDocset))).ToHashSet();
```

Comment:
the key is not Document, is the file path to docset

Reply:
why no test failed?